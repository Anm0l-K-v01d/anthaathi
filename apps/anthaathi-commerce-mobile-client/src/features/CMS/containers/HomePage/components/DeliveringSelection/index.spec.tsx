import React from 'react';
import 'react-native';
import {fireEvent, render} from '@testing-library/react-native';
import {DeliveringSelection} from './index';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';

describe('DeliveringSection', () => {
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <DeliveringSelection
            onPress={() => {}}
            userAddresses={[
              {
                key: 1,
                title: 'Apartment',
                subtitle: '14b street, AI Quoz Industrial Area 4',
              },
              {
                key: 2,
                title: 'Apartment',
                subtitle: '1A street, Discovery Gardens',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('deliveringSelection')).toBeTruthy();
  });

  it('should when we call tap', function () {
    const onpress = jest.fn();

    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <DeliveringSelection
            onPress={onpress}
            userAddresses={[
              {
                key: 1,
                title: 'Apartment',
                subtitle: '14b street, AI Quoz Industrial Area 4',
              },
              {
                key: 2,
                title: 'Apartment',
                subtitle: '1A street, Discovery Gardens',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    fireEvent.press(temp.queryByTestId('deliveringSelection')!);
    expect(onpress).toBeCalledTimes(1);
  });

  it('should have DeliveringSelection country, location name', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <DeliveringSelection
            userAddresses={[
              {
                key: 1,
                title: 'Apartment',
                subtitle: '14b street, AI Quoz Industrial Area 4',
              },
              {
                key: 2,
                title: 'Apartment',
                subtitle: '1A street, Discovery Gardens',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryByText('Delivering to')).toBeTruthy();
  });
});
