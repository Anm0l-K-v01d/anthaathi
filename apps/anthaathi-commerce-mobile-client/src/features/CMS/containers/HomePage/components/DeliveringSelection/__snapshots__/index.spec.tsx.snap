// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeliveringSection should render the item 1`] = `
<View
  testID="deliveringSelection"
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "marginVertical": 5,
      }
    }
  >
    <View
      style={
        [
          {
            "borderRadius": 12,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 6,
            "paddingVertical": 5,
          },
          {
            "backgroundColor": undefined,
            "borderColor": "#6200ee",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 1,
          },
        ]
      }
    >
      <View
        style={
          {
            "alignContent": "center",
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
            "width": 38,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#6200ee",
                "fontSize": 24,
              },
              undefined,
              {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          󰍎
        </Text>
      </View>
      <View
        style={
          {
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              {
                "textAlign": "left",
              },
              undefined,
            ]
          }
          variant="labelMedium"
        >
          Delivering to
        </Text>
        <Text
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              {
                "textAlign": "left",
              },
              undefined,
            ]
          }
          variant="labelLarge"
        >
          Apartment
        </Text>
        <Text
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              {
                "textAlign": "left",
              },
              {
                "fontSize": 12,
              },
            ]
          }
          variant="labelLarge"
        >
          14b street, AI Quoz Industrial Area 4
        </Text>
      </View>
      <View
        style={
          {
            "flexGrow": 1,
          }
        }
      />
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        hitSlop={
          {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            [
              {
                "alignItems": "center",
                "justifyContent": "center",
                "margin": 6,
                "overflow": "hidden",
              },
              {
                "borderRadius": 18,
                "height": 36,
                "width": 36,
              },
              undefined,
              undefined,
            ],
          ]
        }
      >
        <View>
          <Text
            accessibilityElementsHidden={true}
            allowFontScaling={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              [
                {
                  "color": "#000000",
                  "fontSize": 24,
                },
                [
                  {
                    "lineHeight": 24,
                    "transform": [
                      {
                        "scaleX": 1,
                      },
                    ],
                  },
                  {
                    "backgroundColor": "transparent",
                  },
                ],
                {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            󰅂
          </Text>
        </View>
      </View>
    </View>
  </View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  />
</View>
`;
